app/models/ncr/work_order.rb

```diff
@@ -114,13 +114,23 @@ def approving_official
       self.approvers.first
     end
 
-    def current_approver_email_address
+    def current_approver
       if self.pending?
-        self.currently_awaiting_approvers.first.email_address
+        self.currently_awaiting_approvers.first
       elsif self.approving_official
-        self.approving_official.email_address
+        self.approving_official
+      elsif self.emergency and self.approvers.empty?
+        nil
       else
-        self.system_approver_emails.first
+        User.for_email(self.system_approver_emails.first)
+      end
+    end
+
+    def final_approver
+      if self.emergency and self.approvers.empty?
```

Comment:
I find `unless` to be the hardest to understand. I think the easiest way to explain my thinking around this is to explain how I would say each thing in English:


v1: `if self.emergency and self.approvers.empty?`

= "if it is an emergency and there are no approvers, don't sent to anyone. otherwise, send to the last approver"

(in this case, not sure what "otherwise" means without thinking really hard about it)

v2 `if not_emergency? && approvers.any?`

= "if it's not an emergency, and there are some approvers, send it to the last approver"

(pretty easy to understand)

v3 `unless emergency and approvers.empty?`

= "if it is not an emergency and there are no approvers, return the last approver`

(WAIT, that isn't even right in terms of what Ruby will return, but that is how it reads to me)

Given the above interpretations, I think the second is the easiest to understand. But interested in others' opinions as well!